plugins/UM3NetworkPrinting/resources/qml/PrintJobInfoBlock.qml

```diff
@@ -2,13 +2,119 @@ import QtQuick 2.2
+            topLine = catalog.i18nc("@label", "The assigned printer, %1, requires the following configuration change(s):").arg(printJob.assignedPrinter.name);
+        } else {
+            topLine = catalog.i18nc("@label", "The printer %1 is assigned, but the job contains an unknown material configuration.").arg(printJob.assignedPrinter.name);
+        }
+        var result = "<p>" + topLine +"</p>";
+
+        for (var i=0; i<printJob.configurationChanges.length; i++) {
+            var change = printJob.configurationChanges[i];
+            var text = "";
+            if (change.typeOfChange === 'material_change') {
+                text = catalog.i18nc("@label", "Change material %1 from %2 to %3.").arg(change.index + 1).arg(change.originName).arg(change.targetName);
+            } else if (change.typeOfChange === 'material_insert') {
+                text = catalog.i18nc("@label", "Load %3 as material %1 (This cannot be overridden).").arg(change.index + 1).arg(change.targetName);
+            } else if (change.typeOfChange === 'print_core_change') {
+                text = catalog.i18nc("@label", "Change print core %1 from %2 to %3.").arg(change.index + 1).arg(change.originName).arg(change.targetName);
+            } else if (change.typeOfChange === 'buildplate_change') {
+                var target_name = formatBuildPlateType(change.target_name);
+                text = catalog.i18nc("@label", "Change build plate to %1 (This cannot be overridden).").arg(target_name);
+            }
+            result += "<p><b>" + text + "</b></p>"; 
+        }
+        return result;
+    }
+
+    function materialsAreKnown(printJob) {
+        var conf0 = printJob.configuration[0];
+        if (conf0 && !conf0.material.material) {
+            return false;
+        }
+
+        var conf1 = printJob.configuration[1];
+        if (conf1 && !conf1.material.material) {
+            return false;
+        }
+
+        return true;
+    }
+
+    function formatBuildPlateType(buildPlateType) {
+        var translationText = "";
+
+        switch (buildPlateType) {
+            case 'glass':
+                translationText = catalog.i18nc("@label", "Glass");
+                break;
+            case 'aluminum':
+                translationText = catalog.i18nc("@label", "Aluminum");
+                break;
+            default:
+                translationText = null;
+        }
+        return translationText;
+    }
+
+    function formatPrintJobName(name) {
+        var extensionsToRemove = [
+            '.gz',
+            '.gcode',
+            '.ufp'
+        ];
+
+        for (var i = 0; i < extensionsToRemove.length; i++) {
+            var extension = extensionsToRemove[i];
+
+            if (name.slice(-extension.length) === extension) {
+                name = name.substring(0, name.length - extension.length);
+            }
+        }
+
+        return name;
+    }
+
+    function isPrintJobForcable(printJob) {
```

Comment:
Why is this logic in the QML?

Reply:
I didn't see much point in rewriting it for something which is going to thrown away and replaced in a month or two.

Reply:
But historically, hasn't that been so for all of Cura?
(ducks for cover)

Reply:
It's not going to be replaced in a month or two as far as I know. 

Right now, I don't think these changes should be put in 3.5, since they add new features to the printJobOutput model.

Reply:
This is already being fixed in my branch from this branch where I'm rewriting the Print Info Block from scratch to be much cleaner. Don't spend more time reviewing this for now.